control/grpc.py

```diff
@@ -1037,107 +1037,149 @@ def namespace_change_load_balancing_group_safe(self, request, context):
+                except Exception as ex:
+                    errmsg = f"Failure changing load balancing group for namespace {nsid_msg}in {request.subsystem_nqn}. Can't get namespace entry from local state"
+                    self.logger.error(errmsg)
+                    errmsg = f"{errmsg}:\n{ex}"
+                    return pb2.req_status(status=errno.ENOENT, error_message=errmsg)
+            elif request.uuid:
+                for key, val in state.items():
+                    if not key.startswith(self.gateway_state.local.NAMESPACE_PREFIX):
+                        continue
+                    try:
+                        ns = json.loads(val)
+                        if ns["uuid"] == request.uuid:
+                            ns_entry = ns
+                            break
+                    except Exception:
+                        self.logger.exception(f"Got exception trying to get subsystem {nqn} namespaces")
+                        pass
+            else:
+                errmsg = f"Failure changing load balancing group for namespace {nsid_msg}in {request.subsystem_nqn}. At least one of NSID or UUID should be specified"
                 self.logger.error(errmsg)
-                return pb2.req_status(status=errno.ENODEV, error_message=errmsg)
+                return pb2.req_status(status=errno.EINVAL, error_message=errmsg)
 
-            if request.uuid and request.uuid != uuid:
-                errmsg = f"Failure changing load balancing group for namespace {nsid_msg}in {request.subsystem_nqn}: Returned UUID {uuid} differs from requested one {request.uuid}"
+            if not ns_entry:
+                errmsg = f"Failure changing load balancing group for namespace {nsid_msg}in {request.subsystem_nqn}. Can't get namespace entry from local state"
                 self.logger.error(errmsg)
-                return pb2.req_status(status=errno.ENODEV, error_message=errmsg)
+                return pb2.req_status(status=errno.ENOENT, error_message=errmsg)
 
-            bdev_name = find_ret[1]
-            if not bdev_name:
-                bdev_name = self.find_unique_bdev_name(uuid)
-                self.logger.warning(f"Failure finding namespace's associated block device name, will use {bdev_name} instead")
+            if not request.nsid:
+                try:
+                    request.nsid = ns_entry["nsid"]
+                except Exception:
+                    errmsg = f"Failure changing load balancing group for namespace {nsid_msg}in {request.subsystem_nqn}. Can't get namespace NSID"
+                    self.logger.error(errmsg)
+                    return pb2.req_status(status=errno.ENOENT, error_message=errmsg)
 
-            ns_entry = None
-            state = self.gateway_state.local.get_state()
-            ns_key = GatewayState.build_namespace_key(request.subsystem_nqn, nsid)
+            create_image = False
+            ns_size = None
+            force = False
             try:
```

Comment:
@caroav whenever I get attributes from a dictionary I use a try block. In case we somehow don't have these attributes. In that case I'll just use the default values. So, if I can't read the create_image value I will not crash but will try to continue. In the worst case I will fail later if the image is not already there.